data/org.gnome.settings-daemon.plugins.power.gschema.xml.in.in

```diff
@@ -50,5 +50,10 @@
       <summary>Power button action</summary>
       <description>The action to take when the system power button is pressed. This action is hard-coded (and the setting ignored) on virtual machines (power off) and tablets (suspend).</description>
     </key>
+    <key name="lid-closed" type="b">
+      <default>false</default>
+      <summary>Power button action</summary>
+      <description>Whether or not a laptop lid is closed at a given moment. Systems without a lid (desktops, tablets etc) will allways report this as false.</description>
+    </key>
```

Comment:
This is the wrong approach: GSettings are meant to be used for user preferences, not to keep a runtime state.
UPower already keeps the runtime state; you should be able to just create another `UpClient` in the media keys plugin and fetch the state like it's done here: https://github.com/endlessm/gnome-settings-daemon/blob/master/plugins/power/gsd-power-manager.c#L2618